lib/ng/edit/ogc/module.js

```diff
@@ -0,0 +1,460 @@
+      var view = this.map_.getView();
+      this.map_.beforeRender(ol.animation.pan({
+        duration: this.animationDuration_,
+        source: view.getCenter()
+      }));
+      view.setCenter(center);
+      this.map_.beforeRender(ol.animation.zoom({
+        resolution: view.getResolution(),
+        duration: this.animationDuration_
+      }));
+      view.setZoom(zoom);
+    };
+
+    StoryMap.prototype.setAllowPan = function(allowPan) {
+      this.map_.getInteractions().forEach(function(i) {
+        if (i instanceof ol.interaction.KeyboardPan ||
+          i instanceof ol.interaction.DragPan) {
+            i.setActive(allowPan);
+        }
+      });
+    };
+
+    StoryMap.prototype.setAllowZoom = function(allowZoom) {
+      var zoomCtrl;
+      this.map_.getControls().forEach(function(c) {
+        if (c instanceof ol.control.Zoom) {
+          zoomCtrl = c;
+        }
+      });
+      if (!allowZoom) {
+        this.map_.removeControl(zoomCtrl);
+      } else {
+        this.map_.addControl(new ol.control.Zoom());
+      }
+      this.map_.getInteractions().forEach(function(i) {
+        if (i instanceof ol.interaction.DoubleClickZoom ||
+          i instanceof ol.interaction.PinchZoom ||
+          i instanceof ol.interaction.DragZoom ||
+          i instanceof ol.interaction.MouseWheelZoom) {
+            i.setActive(allowZoom);
+        }
+      });
+    };
+
+    module.constant('StoryMap', StoryMap);
+
+    function StoryLayer(data) {
+      if (data.times && storytools.core.time.utils.isRangeLike(data.times)) {
+        data.times = new storytools.core.time.utils.Interval(data.times);
+      }
+      ol.Object.call(this, data);
+      var layer;
+      if (this.get('type') === 'VECTOR') {
+        layer = new ol.layer.Vector();
+      } else if (this.get('type') === 'WMS') {
+        var config = {
+          useOldAsInterimTiles: true
+        };
+        if (this.get('singleTile') === true) {
+          layer = new ol.layer.Image(config);
+        } else {
+          layer = new ol.layer.Tile(config);
+        }
+      }
+      this.layer_ = layer;
+    }
+
+    StoryLayer.prototype = Object.create(ol.Object.prototype);
+    StoryLayer.prototype.constructor = StoryLayer;
+
+    StoryLayer.prototype.getState = function() {
+      return this.getProperties();
+    };
+
+    StoryLayer.prototype.getLayer = function() {
+      return this.layer_;
+    };
+
+    StoryLayer.prototype.setLayer = function(layer) {
+      this.layer_ = layer;
+    };
+
+    function EditableStoryLayer(data) {
+      StoryLayer.call(this, data);
+    }
+
+    EditableStoryLayer.prototype = Object.create(StoryLayer.prototype);
+    EditableStoryLayer.prototype.constructor = EditableStoryLayer;
+
+    module.constant('StoryLayer', StoryLayer);
+    module.constant('EditableStoryLayer', EditableStoryLayer);
+
+    module.service('stAnnotateLayer', function($http, $q) {
+      return {
+        setWMSSource: function(EditableStoryLayer) {
```

Comment:
not a major thing but this threw me off at first as the argument name is the same as the class and looks like it might be getting injected (just my convention - I know injection is not possible here). Suggestion: change to camel case or just layer.